streamer/periodconcat_node.py

```diff
@@ -38,6 +41,39 @@ def __init__(self,
     self._pipeline_config = pipeline_config
     self._output_dir = output_dir
     self._packager_nodes: List[PackagerNode] = packager_nodes
+    
+    # know whether the first period has video and audio or not.
+    fp_has_vid, fp_has_aud = False, False
+    for output_stream in packager_nodes[0].output_streams:
+      if isinstance(output_stream, VideoOutputStream):
+        fp_has_vid = True
+      elif isinstance(output_stream, AudioOutputStream):
+        fp_has_aud = True
+    
+    for i, packager_node in enumerate(self._packager_nodes):
+      has_vid, has_aud = False, False
+      for output_stream in packager_node.output_streams:
+        if isinstance(output_stream, VideoOutputStream):
+          has_vid = True
+        elif isinstance(output_stream, AudioOutputStream):
+          has_aud = True
+      if has_vid != fp_has_vid or has_aud != fp_has_aud:
+        # Overwrite the start and the stop methods.
+        setattr(self, 'start', lambda: None)
+        setattr(self, 'stop', lambda _=None: None)
+        print("\nWARNING: Stopping period concatenation.")
+        print("Period#{} has {}video and has {}audio while Period#1 "
+              "has {}video and has {}audio.".format(i + 1, 
+                                                    "" if has_vid else "no ",
+                                                    "" if has_aud else "no ",
+                                                    "" if fp_has_vid else "no ",
+                                                    "" if fp_has_aud else "no "))
+        print("\nHINT:\n\tBe sure that either all the periods have video or all do not,\n"
+              "\tand all the periods have audio or all do not, i.e. don't mix videoless\n"
+              "\tperiods with other periods that have video that is for the concatenation\n"
+              "\tto be performed successfully.\n")
```

Comment:
English nit:  This sentence is too long.  Please end this sentence after "other periods that have video."  If you want to say that this is necessary for concatenation, make that a second sentence.